### PR TITLE
[DOCS] IDs for doc snippets 

### PR DESCRIPTION
**Pull request by probakowski**
This PR adds IDs for examples in _reference/aggregations/bucket_ and _reference/aggregations/matrix_ 
This make test names more stable so pure docs changes in that files won't trigger tests (they will be cached).
It also ignores _build-idea_ folder when looking for snippets (which is used by my IntelliJ when running gradle tasks from it for some reason)
Relates to #46951

### CODE REVIEW

**Review by nik9000 — APPROVED** [review]
:metal:

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/doc/SnippetsTask.groovy [hunk 1]
```diff
@@ -66,6 +66,7 @@ class SnippetsTask extends DefaultTask {
         exclude 'build.gradle'
         // That is where the snippets go, not where they come from!
         exclude 'build'
+        exclude 'build-idea'
```
Comment (atorok) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/doc/SnippetsTask.groovy]: there's also `build-eclipse`. The reason we have these is to make sure builds from the IDE don't interfere with those on the command line.

**Review by alpar-t — APPROVED** [review]
Thanks!

**Comment by probakowski** [comment]
@elasticmachine update branch

**Pull request merged**